From 4aa052224404dec656968fa9e35685433c05265e Mon Sep 17 00:00:00 2001
From: Jianqun xu <jay.xu@rock-chips.com>
Date: Thu, 17 Dec 2015 18:33:51 +0800
Subject: [PATCH] ARM64: dts: rockchip: add dts for RK3368 Sheep board

The Rockchip rk3368 top board maybe match to several types of screen,
such as LVDS, MIPI and EDP. It's confuse with current dts file.

This patch to add a dtsi file - rk3368-tb.dtsi, put the common nodes
and set their status to be "okay". But for the screen type, it hasn't
enable any type of screen, dts file should do it.

For example, the patch adds rk3368-tp-sheep.dts for Sheep board, which
uses mipi screen with rk818 pmic.

Change-Id: Ide5952552283a078f4eb7368507a7cf553275021
Signed-off-by: Jianqun xu <jay.xu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/Makefile         |  2 +-
 .../boot/dts/rockchip/rk3368-tb-sheep.dts     | 53 +++++++++++++++++++
 .../{rk3368-tb.dts => rk3368-tb.dtsi}         | 28 +++++-----
 arch/arm64/boot/dts/rockchip/rk3368.dtsi      | 28 ++++++----
 4 files changed, 87 insertions(+), 24 deletions(-)
 create mode 100644 arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
 rename arch/arm64/boot/dts/rockchip/{rk3368-tb.dts => rk3368-tb.dtsi} (96%)

diff --git a/arch/arm64/boot/dts/rockchip/Makefile b/arch/arm64/boot/dts/rockchip/Makefile
index 5ae2a8c3cad3..17a1c9359833 100644
--- a/arch/arm64/boot/dts/rockchip/Makefile
+++ b/arch/arm64/boot/dts/rockchip/Makefile
@@ -1,5 +1,5 @@
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3368-r88.dtb
-dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3368-tb.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3368-tb-sheep.dtb
 
 always		:= $(dtb-y)
 subdir-y	:= $(dts-dirs)
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts b/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
new file mode 100644
index 000000000000..d66474875f80
--- /dev/null
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb-sheep.dts
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2015 Fuzhou Rockchip Electronics Co., Ltd
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPL or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This file is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This file is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/dts-v1/;
+#include "rk3368-tb.dtsi"
+
+/ {
+	model = "Rockchip Sheep board";
+	compatible = "rockchip,sheep", "rockchip,rk3368";
+};
+
+&syr827 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/rockchip/rk3368-tb.dts b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
similarity index 96%
rename from arch/arm64/boot/dts/rockchip/rk3368-tb.dts
rename to arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
index b562ed1021ff..686a874b1d03 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-tb.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-tb.dtsi
@@ -40,12 +40,10 @@
  *     OTHER DEALINGS IN THE SOFTWARE.
  */
 
-/dts-v1/;
 #include <dt-bindings/pwm/pwm.h>
 #include "rk3368.dtsi"
 
 / {
-	model = "Rockchip Top board";
 	compatible = "rockchip,tb", "rockchip,rk3368";
 
 	chosen {
@@ -117,10 +115,11 @@
 
 &i2c0 {
 	status = "okay";
+
 	syr827: syr827@40 {
 		compatible = "silergy,syr82x";
+		status = "disabled";
 		reg = <0x40>;
-		status = "okay";
 		regulators {
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -144,8 +143,8 @@
 
 	syr828: syr828@41 {
 		compatible = "silergy,syr82x";
+		status = "disabled";
 		reg = <0x41>;
-		status = "okay";
 		regulators {
 			#address-cells = <1>;
 			#size-cells = <0>;
@@ -173,8 +172,8 @@
 		 * it's better to add both instances here.
 		 */
 		compatible = "act,act8846", "active-semi,act8846";
+		status = "disabled";
 		reg = <0x5a>;
-		status = "okay";
 		system-power-controller;
 
 		regulators {
@@ -274,15 +273,11 @@
 };
 
 &fb {
+	status = "okay";
 	rockchip,disp-mode = <NO_DUAL>;
 	rockchip,uboot-logo-on = <0>;
 };
 
-&rk_screen {
-	/*#include <dt-bindings/display/screen-timing/lcd-b101ew05.dtsi>*/
-	#include <dt-bindings/display/screen-timing/lcd-F402.dtsi>
-};
-
 &lcdc {
 	status = "okay";
 	backlight = <&backlight>;
@@ -291,19 +286,19 @@
 	rockchip,fb-win-map = <FB_DEFAULT_ORDER>;
 	power_ctr: power_ctr {
 		rockchip,debug = <0>;
-		lcd_en:lcd_en {
+		lcd_en: lcd-en {
 			rockchip,power_type = <GPIO>;
 			gpios = <&gpio0 22 GPIO_ACTIVE_HIGH>;/*GPIO_C6 = 22*/
 			rockchip,delay = <10>;
 		};
 
-		lcd_cs:lcd_cs {
+		lcd_cs: lcd-cs {
 			rockchip,power_type = <GPIO>;
 			gpios = <&gpio0 21 GPIO_ACTIVE_HIGH>;/*GPIO_C5 = 21*/
 			rockchip,delay = <10>;
 		};
 
-		/*lcd_rst:lcd_rst {
+		/*lcd_rst: lcd-rst {
 			rockchip,power_type = <GPIO>;
 			gpios = <&gpio3 GPIO_D6 GPIO_ACTIVE_HIGH>;
 			rockchip,delay = <5>;
@@ -312,8 +307,13 @@
 };
 
 &lvds {
-	status = "disabled";
 	pinctrl-names = "lcdc", "sleep";
 	pinctrl-0 = <&lcdc_lcdc>;
 	pinctrl-1 = <&lcdc_gpio>;
+	status = "disabled";
 };
+
+&vopb_mmu {
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/rockchip/rk3368.dtsi b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
index e494eb768e5f..a433fbd0feac 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3368.dtsi
@@ -1092,10 +1092,12 @@
 	fb: fb {
 		compatible = "rockchip,rk-fb";
 		rockchip,disp-mode = <NO_DUAL>;
+		status = "disabled";
 	};
 
-	rk_screen: rk_screen {
+	rk_screen: screen {
 		compatible = "rockchip,screen";
+		status = "disabled";
 	};
 
 	lcdc: lcdc@ff930000 {
@@ -1108,12 +1110,12 @@
 		rockchip,iommu-enabled = <1>;
 		reg = <0x0 0xff930000 0x0 0x10000>;
 		interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
-		status = "disabled";
 		clocks = <&cru ACLK_VOP>, <&cru DCLK_VOP>, <&cru HCLK_VOP>;
 		clock-names = "aclk_lcdc", "dclk_lcdc", "hclk_lcdc";
 		/*power-domains = <&power PD_VIO>;*/
 		resets = <&cru SRST_LCDC0_AXI>, <&cru SRST_LCDC0_AHB>, <&cru SRST_LCDC0_DCLK>;
 		reset-names = "axi", "ahb", "dclk";
+		status = "disabled";
 	};
 
 	lvds: lvds@ff968000 {
@@ -1137,64 +1139,72 @@
 		/*power-domains = <&power PD_VIO>;*/
 		resets = <&cru SRST_EDP_24M>, <&cru SRST_EDP>;
 		reset-names = "edp_24m", "edp_apb";
+		status = "disabled";
 	};
 
-	iep_mmu {
+	iep_mmu: iep-mmu {
 		dbgname = "iep";
 		compatible = "rockchip,iep_mmu";
 		reg = <0x0 0xff900800 0x0 0x100>;
 		interrupts = <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "iep_mmu";
+		status = "disabled";
 	};
 
-	vip_mmu {
+	vip_mmu: vip-mmu {
 		dbgname = "vip";
 		compatible = "rockchip,vip_mmu";
 		reg = <0x0 0xff950800 0x0 0x100>;
 		interrupts = <GIC_SPI 13 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "vip_mmu";
+		status = "disabled";
 	};
 
-	vop_mmu {
+	vopb_mmu: vopb-mmu {
 		dbgname = "vop";
 		compatible = "rockchip,vopb_mmu";
 		reg = <0x0 0xff930300 0x0 0x100>;
 		interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "vop_mmu";
+		status = "disabled";
 	};
 
-	isp_mmu {
+	isp_mmu: isp-mmu {
 		dbgname = "isp_mmu";
 		compatible = "rockchip,isp_mmu";
 		reg = <0x0 0xff914000 0x0 0x100>,
 		      <0x0 0xff915000 0x0 0x100>;
 		interrupts = <GIC_SPI 14 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "isp_mmu";
+		status = "disabled";
 	};
 
-	hdcp_mmu {
+	hdcp_mmu: hdcp-mmu {
 		 dbgname = "hdcp_mmu";
 		 compatible = "rockchip,hdcp_mmu";
 		 reg = <0x0 0xff940000 0x0 0x100>;
 		 interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
 		 interrupt-names = "hdcp_mmu";
+		status = "disabled";
 	};
 
-	hevc_mmu {
+	hevc_mmu: hevc-mmu {
 		dbgname = "hevc";
 		compatible = "rockchip,hevc_mmu";
 		reg = <0x0 0xff9a0440 0x0 0x40>,
 		      <0x0 0xff9a0480 0x0 0x40>;
 		interrupts = <GIC_SPI 12 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "hevc_mmu";
+		status = "disabled";
 	};
 
-	vpu_mmu {
+	vpu_mmu: vpu-mmu {
 		dbgname = "vpu";
 		compatible = "rockchip,vpu_mmu";
 		reg = <0x0 0xff9a0800 0x0 0x100>;
 		interrupts = <GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>,
 			     <GIC_SPI 10 IRQ_TYPE_LEVEL_HIGH>;
 		interrupt-names = "vepu_mmu", "vdpu_mmu";
+		status = "disabled";
 	};
 };
-- 
2.35.3

